semantic/check_test.go

```diff
@@ -1216,3 +1338,960 @@ func TestVarAssignmentChecks(t *testing.T) {
+												Body: []Statement{
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "a"},
+														Init:       &IntegerLiteral{Value: 1},
+													},
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "g"},
+														Init: &FunctionExpression{
+															Block: &FunctionBlock{
+																Parameters: &FunctionParameters{
+																	List: []*FunctionParameter{
+																		{
+																			Key: &Identifier{Name: "g"},
+																		},
+																	},
+																},
+																Body: &CallExpression{
+																	Callee: &IdentifierExpression{Name: "f"},
+																	Pipe:   &IdentifierExpression{Name: "g"},
+																},
+															},
+														},
+													},
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "h"},
+														Init: &FunctionExpression{
+															Block: &FunctionBlock{
+																Parameters: &FunctionParameters{
+																	List: []*FunctionParameter{
+																		{
+																			Key: &Identifier{Name: "b"},
+																		},
+																	},
+																},
+																Body: &CallExpression{
+																	Callee: &IdentifierExpression{Name: "g"},
+																	Arguments: &ObjectExpression{
+																		Properties: []*Property{
+																			{
+																				Key:   &Identifier{Name: "g"},
+																				Value: &IdentifierExpression{Name: "b"},
+																			},
+																		},
+																	},
+																},
+															},
+														},
+													},
+													&ReturnStatement{
+														Argument: &CallExpression{
+															Callee: &IdentifierExpression{Name: "h"},
+															Arguments: &ObjectExpression{
+																Properties: []*Property{
+																	{
+																		Key:   &Identifier{Name: "b"},
+																		Value: &IdentifierExpression{Name: "a"},
+																	},
+																},
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						Package: &PackageClause{
+							Name: &Identifier{Name: "foo"},
+						},
+						Body: []Statement{
+							&OptionStatement{
+								Assignment: &NativeVariableAssignment{
+									Identifier: &Identifier{Name: "b"},
+									Init: &IdentifierExpression{
+										loc: loc{
+											File: "b.flux",
+											Start: ast.Position{
+												Line:   2,
+												Column: 12,
+											},
+											End: ast.Position{
+												Line:   2,
+												Column: 13,
+											},
+										},
+										Name: "a",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err: fmt.Errorf(`option dependency: option "b" depends on option "a" defined in the same package at b.flux|2:12-2:13`),
```

Comment:
Is this possible to do?
```
package p
option o1 = 20
```
```
package q
import "p"
option p.o1 = p.o1 * 2
```
```
package r
import "p"
option p.o1 = p.or + 2
```
```
package s
import "p"
import "r"
import "s"
... p.o1 ... // use of o1
```
(o1 could be either 44 or 42)

Reply:
Actually I didn't see any package qualifiers on the LHS of option assignments here, is that possible to do?

Reply:
Yes reassigning options declared in an imported package is allowed. And in your example `o1` would be 44 because imports are ordered within a file. However there is no order across files. Hence if two files in the same package both imported other packages that mutated the same option, this would be non-deterministic behavior and is not allowed.